### PR TITLE
Add styleURI and cameraOptions to MapInitOptions

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [X] Write tests for all new functionality. If tests were not written, please explain why.
 - [X] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [X] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: 
    - <changelog>Added `StyleURI` and `CameraOptions` parameters to `MapInitOptions`; removed `StyleURI` from map view initializers.</changelog>.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR:

- Adds an initial `StyleURI` and `CameraOptions` to the `MapInitOptions`.
- Removes the `StyleURI` parameters from `MapView.init(...)` and `BaseMapView.init(...)`
- Fixes a bug with GlyphRasterizationOptions (where a font family would be provided if mode == `.noGlyphsRasterizedLocally`

~Waiting for camera PR to land before this can.~

### CODE REVIEW

File: Tests/MapboxMapsTests/MapView/Integration Tests/Map/MapInitOptionsIntegrationTests.swift [hunk 1]
```diff
@@ -89,6 +97,70 @@ class MapInitOptionsIntegrationTests: XCTestCase {
...
+
+        let destCenter = view.cameraOptions.center!
+        let destZoom = view.cameraOptions.zoom!
+
+        XCTAssertEqual(sourceCenter[0], destCenter.longitude, accuracy: 0.0000001)
+        XCTAssertEqual(sourceCenter[1], destCenter.latitude, accuracy: 0.0000001)
+        XCTAssertEqual(sourceZoom, destZoom, accuracy: 0.0000001)
+    }
+
+    func testInitialCameraOverridesStyleDefaultCamera() throws {
```
Comment (nishant-karajgikar) [inline Tests/MapboxMapsTests/MapView/Integration Tests/Map/MapInitOptionsIntegrationTests.swift:136]: Nice test!

File: Sources/MapboxMaps/Style/Style.swift [hunk 2]
```diff
@@ -5,9 +5,19 @@ import MapboxMapsFoundation
 
 //swiftlint:disable file_length
 public class Style {
+
+    public static let defaultURI = StyleURI.streets
```
Comment (nishant-karajgikar) [inline Sources/MapboxMaps/Style/Style.swift:9]: does this need to be public?

**Review by nishant-karajgikar — APPROVED** [review]
LGTM.. A couple comments but no blockers.

File: Sources/MapboxMaps/Style/Style.swift (hunk 2 above)
Comment (julianrex) [reply Sources/MapboxMaps/Style/Style.swift:9]: I think it does since it's used as a default value.

**Review by jmkiley — APPROVED** [review]

**Pull request merged**